File: java/LinkedList/delete.java
import java.util.LinkedList;
import java.util.Scanner;

public class LinkedListDelete {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        LinkedList<Integer> list = new LinkedList<>();

        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();

        System.out.println("Enter elements:");
        for (int i = 0; i < n; i++) list.add(sc.nextInt());

        System.out.println("Original LinkedList: " + list);

        // Delete first
        list.removeFirst();
        // Delete last
        list.removeLast();

        // Delete at specific position
        System.out.print("Enter position to delete (0-based): ");
        int pos = sc.nextInt();
        if (pos >= 0 && pos < list.size()) list.remove(pos);

        System.out.println("LinkedList after deletions: " + list);
    }
}

File: python/LinkedList/delete.py
class LinkedList(LinkedList):
    def remove_first(self):
        if self.head:
            self.head = self.head.next

    def remove_last(self):
        if not self.head:
            return
        if not self.head.next:
            self.head = None
            return
        curr = self.head
        while curr.next.next:
            curr = curr.next
        curr.next = None

    def remove_at_position(self, pos):
        if pos == 0:
            self.remove_first()
            return
        curr = self.head
        for _ in range(pos-1):
            if not curr.next:
                print("Position out of bounds")
                return
            curr = curr.next
        if curr.next:
            curr.next = curr.next.next

# Driver
ll = LinkedList()
n = int(input("Enter number of elements: "))
for _ in range(n):
    ll.append(int(input()))

ll.remove_first()
ll.remove_last()

pos = int(input("Enter position to remove: "))
ll.remove_at_position(pos)

print("LinkedList after deletions:")
ll.display()
